### PR TITLE
adding basic files for having preferences in a different db

### PR DESCRIPTION
**Pull request by nmonterroso**
@Wikia/services-team 

laying out the foundation to talk to a separate database for a lot of the work to be done in SERVICES-548.

### CODE REVIEW

File: service/user-preference/src/main/java/com/wikia/userpreference/domain/Preference.java [hunk 1]
```diff
@@ -14,8 +14,25 @@
 @Data
```
Comment (WikiaTech) [inline service/user-preference/src/main/java/com/wikia/userpreference/domain/Preference.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: service/user-preference/src/main/java/com/wikia/userpreference/dao/UserPreferencesDAO.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+import com.google.common.collect.ImmutableList;
+import org.jooq.Condition;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.SelectConditionStep;
+
+import java.nio.charset.Charset;
+import java.sql.SQLException;
+import java.util.Collections;
```
Comment (WikiaTech) [inline service/user-preference/src/main/java/com/wikia/userpreference/dao/UserPreferencesDAO.java]: MINOR Remove this unused import 'java.util.Collections'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

File: service/user-preference/src/main/java/com/wikia/userpreference/dao/UserPreferencesDAO.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
```
Comment (WikiaTech) [inline service/user-preference/src/main/java/com/wikia/userpreference/dao/UserPreferencesDAO.java]: MINOR Remove this unused import 'java.util.regex.Pattern'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

File: service/user-preference/src/main/java/com/wikia/userpreference/dao/UserPreferencesDAO.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  @Override
+  public boolean updatePreferences(int userId, List<Preference> preferences) throws SQLException {
+    return false;
+  }
+
+  @Override
+  public boolean setPreferences(int userId, List<Preference> preferences) throws SQLException {
+    return false; // should be a transaction start, delete preferences, update preferences
```
Comment (WikiaTech) [inline service/user-preference/src/main/java/com/wikia/userpreference/dao/UserPreferencesDAO.java]: MINOR Move this trailing comment on the previous empty line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ATrailingCommentCheck)

File: service/user-preference/src/main/java/com/wikia/userpreference/dao/UserPreferencesDAO.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+import static com.wikia.jooq.generated.user_preferences.UserPreferences.USER_PREFERENCES;
+import static com.wikia.jooq.generated.user_preferences.Tables.USER_PREFERENCE;
+
+import com.wikia.dropwizard.databases.jooq.JooqConfigs;
+import com.wikia.userpreference.domain.Preference;
+import com.wikia.userpreference.domain.Whitelist;
+
+import com.google.common.collect.ImmutableList;
+import org.jooq.Condition;
```
Comment (WikiaTech) [inline service/user-preference/src/main/java/com/wikia/userpreference/dao/UserPreferencesDAO.java]: MINOR Remove this unused import 'org.jooq.Condition'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues:
* MINOR 5 minor

Watch the comments in this conversation to review them.

**Comment by nmonterroso** [comment]
Jenkins seems to be very unreliable here. Locally the tests pass.

**Comment by drsnyder** [comment]
lgtm

File: service/user-preference/src/main/java/com/wikia/userpreference/domain/Preference.java [hunk 6]
```diff
@@ -10,12 +10,27 @@
 @AllArgsConstructor
 @NoArgsConstructor
 
-public
-@Data
-class Preference {
+public @Data class Preference {
```
Comment (WikiaTech) [inline service/user-preference/src/main/java/com/wikia/userpreference/domain/Preference.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Pull request merged**